app/models/plugin.rb

```diff
@@ -38,4 +38,14 @@ def self.loaded
   def self.capability_category_to_plugin_name( capability_category )
     capability_category.to_s.sub( %r{_[a-z]+$}, '' ).classify.pluralize
   end
+
+  def self.base_records
+    return @base_records if @base_records
+
+    @base_records = []
+    loaded.each do |plugin|
+      base_records << plugin.constantize::ApplicationRecord
```

Comment:
Is `base_records` call on line 47 correct? Looks like it'll call the `.base_records` method recursively.